skiko/src/nativeInterop/cinterop/skia.def

```diff
@@ -0,0 +1,26 @@
+package = org.jetbrains.skiko.skia.native
+language = C++
+plugin = org.jetbrains.kotlin.native.cinterop.plugin.skia
+compilerOpts.osx = -x c++ -std=c++14 \
+    -DSK_GL \
+    -DSK_SUPPORT_GPU=1 \
+    -DSK_SHAPER_CORETEXT_AVAILABLE \
+    -DSK_BUILD_FOR_MAC \
+    -DSK_METAL \
+    -DSK_ALLOW_STATIC_GLOBAL_INITIALIZERS=1 \
+    -DSK_FORCE_DISTANCE_FIELD_TEXT=0 \
+    -DSK_GAMMA_APPLY_TO_A8 \
+    -DSK_GAMMA_SRGB \
+    -DSK_SCALAR_TO_FLOAT_EXCLUDED \
+    -DSK_SHAPER_HARFBUZZ_AVAILABLE \
+    -DSK_SUPPORT_OPENCL=0 \
+    -DSK_UNICODE_AVAILABLE
+
+headers = include/core/SkCanvas.h include/core/SkData.h include/core/SkImage.h include/core/SkMath.h include/core/SkStream.h include/core/SkSurface.h include/core/SkSurfaceProps.h include/core/SkTypes.h include/core/SkPaint.h include/core/SkPictureRecorder.h include/core/SkImageInfo.h include/core/SkColor.h include/core/SkColorSpace.h include/gpu/GrDirectContext.h include/gpu/GrBackendSurface.h include/gpu/GrConfig.h include/gpu/gl/GrGLInterface.h
+headerFilter =  include/core/SkCanvas.h include/core/SkData.h include/core/SkImage.h include/core/SkMath.h include/core/SkStream.h include/core/SkSurface.h include/core/SkSurfaceProps.h include/core/SkTypes.h include/core/SkPaint.h include/core/SkPictureRecorder.h include/core/SkImageInfo.h include/core/SkColor.h include/core/SkColorSpace.h include/gpu/GrDirectContext.h include/gpu/GrBackendSurface.h include/gpu/GrConfig.h include/gpu/gl/GrGLInterface.h
+
+# const overloads are not supported yet
+excludedFunctions = saveLayer mapRect ptr priv
+
+linkerOpts.osx = -framework CoreServices -framework CoreText -framework CoreGraphics -framework Metal
```

Comment:
Previous code gives impression that we use OpenGL for rendering, while this code links Metal. What is actually correct?

Reply:
OpenGL is added by K/N automatically. 
While Metal is still needed for libskia.a to resolve symbols.
```
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_MTLRenderPipelineColorAttachmentDescriptor", referenced from:
      objc-class-ref in libskia.a(gpu.GrMtlPipelineStateBuilder.o)
  "_OBJC_CLASS_$_MTLRenderPipelineDescriptor", referenced from:
      objc-class-ref in libskia.a(gpu.GrMtlPipelineStateBuilder.o)
  "_OBJC_CLASS_$_MTLVertexDescriptor", referenced from:
      objc-class-ref in libskia.a(gpu.GrMtlPipelineStateBuilder.o)
  "_OBJC_CLASS_$_MTLTextureDescriptor", referenced from:
      objc-class-ref in libskia.a(gpu.GrMtlUtil.o)
      objc-class-ref in libskia.a(gpu.GrMtlGpu.o)
      objc-class-ref in libskia.a(gpu.GrMtlAttachment.o)
      objc-class-ref in libskia.a(gpu.GrMtlRenderTarget.o)
      objc-class-ref in libskia.a(gpu.GrMtlTextureRenderTarget.o)
  "_OBJC_CLASS_$_MTLRenderPassDescriptor", referenced from:
      objc-class-ref in libskia.a(gpu.GrMtlGpu.o)
      objc-class-ref in libskia.a(gpu.GrMtlOpsRenderPass.o)
  "_OBJC_CLASS_$_MTLDepthStencilDescriptor", referenced from:
      objc-class-ref in libskia.a(gpu.GrMtlDepthStencil.o)
  "_OBJC_CLASS_$_MTLStencilDescriptor", referenced from:
      objc-class-ref in libskia.a(gpu.GrMtlDepthStencil.o)
  "_OBJC_CLASS_$_MTLSamplerDescriptor", referenced from:
      objc-class-ref in libskia.a(gpu.GrMtlSampler.o)
ld: symbol(s) not found for architecture x86_64
```

Reply:
OK.